pkg/services/publicdashboards/database/database.go

```diff
@@ -283,15 +283,15 @@ func (d *PublicDashboardStoreImpl) Delete(ctx context.Context, uid string) (int6
 	return affectedRows, err
 }
 
-func (d *PublicDashboardStoreImpl) FindByDashboardFolder(ctx context.Context, dashboard *dashboards.Dashboard) ([]*PublicDashboard, error) {
-	if dashboard == nil || !dashboard.IsFolder {
+func (d *PublicDashboardStoreImpl) FindByFolder(ctx context.Context, orgId int64, folderUid string) ([]*PublicDashboard, error) {
+	if orgId == 0 || folderUid == "" {
```

Comment:
folderUID is empty string means that it is the root folder, so this clause won't work

Reply:
also maybe this clause can be removed altogether, where the function call happens there is a clause if this is a folder, and from the name `FindByFolder` it insinuates this method should be called by folders, wdyt?

Reply:
Because of the method name `FindByFolder,`, I think it's better to name the argument as `folderUid`, and it is the responsibility of the caller to send the correct `folderUid` if it is empty. There is no need to execute the query, so it's better to return early. The only method using this method is indeed `FindByFolder` and should be called only for folders but this can change in the future and be used in other places so returning early when the parameters are blank could help not execute the database query unnecessarily